File: 1-easy/problems/0821-shortest-distance-to-a-character/solution.ts
function shortestToChar(S: any, C: any): boolean | number | string | any {
  """
  :type S: str
  :type C: str
  :rtype: List[int]
  """
  result = [(S).length] * (S).length
  prev = -(S).length
  for i in itertools.chain(Array.from({length: (S}, (_, i) => i).length),
                           reversed(Array.from({length: (S}, (_, i) => i).length))):
      if (S[i] == C) {
          prev = i
      result[i] = min(result[i], abs(i-prev))
  return result;
}

export { shortestToChar };

File: 1-easy/problems/0821-shortest-distance-to-a-character/solution.py
from typing import List, Optional, Dict, Set, Tuple

class Solution:
    def shortestToChar(self, S, C):
        """
        :type S: str
        :type C: str
        :rtype: List[int]
        """
        result = [len(S)] * len(S)
        prev = -len(S)
        for i in itertools.chain(range(len(S)),
                                 reversed(range(len(S)))):
            if S[i] == C:
                prev = i
            result[i] = min(result[i], abs(i-prev))
        return result
